facilitator/src/batch.rs

```diff
@@ -123,7 +123,7 @@ impl<'a, H: Header, P: Packet> BatchReader<'a, H, P> {
 
     /// Return the parsed header from this batch, but only if its signature is
     /// valid.
-    pub fn header(&self, key: &UnparsedPublicKey<Vec<u8>>) -> Result<H> {
+    pub fn header(&mut self, key: &UnparsedPublicKey<Vec<u8>>) -> Result<H> {
```

Comment:
Why does header need a `mut self` now?

Reply:
The implementation of `Transport::get` in `GCSTransport` has to be `&mut self` because it mutates its `oauth_token_provider` member when it caches an auth token. `Transport::get` should always have taken `&mut self` because it's easy to imagine a reader where doing a read advances a cursor or consumes a stream, so I didn't feel bad about doing that, but Rust will then make you propagate mutness all over the program, kind of like that "is your function red or blue" blog you showed me the other day.

Reply:
Interesting. I don't think you generally have to propagate mutness all over the program in the same way that async leads to. But it looks like this is inherited from https://github.com/abetterinternet/prio-server/pull/88, so I'll finish reviewing that and see if I grok it.